### PR TITLE
#46 - Add rate limit handler

### PR DESCRIPTION
**Pull request by orlowski11**
added rate limit handling and tests

### CODE REVIEW

File: tests/Feature/RateLimiterTest.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+            $this->requestUrl => MockResponse::make(body: "", status: 200),
+        ]);
+
+        $this->githubConnector->withMockClient($mockClient);
+
+        $this->expectException(RateLimitReachedException::class);
+
+        for ($i = 0; $i = 5000; $i++) {
+            $this->githubConnector->send($this->request);
+        }
```
Comment (kamilpiech97) [inline tests/Feature/RateLimiterTest.php]: Let's set a custom rate limiter for testing, a loop with 5000 calls looks ugly for me :vomiting_face:

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: tests/Feature/RateLimiterTest.php (hunk 1 above)
Comment (kamilpiech97) [reply tests/Feature/RateLimiterTest.php]: @krzysztofrewak what do u think?

Comment (krzysztofrewak) [reply tests/Feature/RateLimiterTest.php]: Now it looks better.

File: tests/Feature/RateLimiterTest.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            $this->requestUrl => MockResponse::make(body: "", status: 429),
+        ]);
+
+        $this->githubConnector->withMockClient($mockClient);
+
+        $this->expectException(RateLimitReachedException::class);
+
+        $this->githubConnector->send($this->request);
+
+        Cache::clear();
```
Comment (kamilpiech97) [inline tests/Feature/RateLimiterTest.php:101]:
```suggestion
```

File: app/Http/Integrations/GithubConnector.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        return [
+            "Content-Type" => "application/json",
+            "Accept" => "application/json",
+        ];
+    }
+
+    protected function resolveLimits(): array
+    {
+        return [
+            Limit::allow(5000)->everyHour(),
```
Comment (kamilpiech97) [inline app/Http/Integrations/GithubConnector.php]: Lets move that number into some config file.

File: tests/Feature/RateLimiterTest.php (hunk 2 above)
Comment (kamilpiech97) [reply tests/Feature/RateLimiterTest.php:101]: For what is that cache clear?

File: tests/Feature/RateLimiterTest.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+class GithubConnectorTestVersion extends GithubConnector
+{
+    protected function resolveLimits(): array
+    {
+        return [
+            Limit::allow(0)->everyHour(),
+        ];
+    }
+}
```
Comment (kamilpiech97) [inline tests/Feature/RateLimiterTest.php]: For me its not good way, lets move limit into config and change config value in tests and test them.

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: tests/Feature/RateLimiterTest.php (hunk 2 above)
Comment (orlowski11) [reply tests/Feature/RateLimiterTest.php:101]: to clear request hits

File: app/Http/Integrations/GithubConnector.php (hunk 3 above)
Comment (orlowski11) [reply app/Http/Integrations/GithubConnector.php]: moved limit to config

File: tests/Feature/RateLimiterTest.php (hunk 4 above)
Comment (orlowski11) [reply tests/Feature/RateLimiterTest.php]: moved limit to config and changing it for tests

**Review by kamilpiech97 — APPROVED** [review]

**Pull request merged**